### ISSUE TITLE
WSL2 clock skew

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#1166]
Some of our Cadence unit tests are failing sometimes due to apparent clock skew between the WSL Linux distro and the host machine.  This appears to be known issue:

https://github.com/microsoft/WSL/issues/5324

It looks like the time can be way off after the host machine has hibernated (although this apparently has been fixed).  Clock skew still looks like it can accumulate over time, which is what I think we're seeing.

I don't believe this will be a problem for our WSL2 based neonKUBE distro because we're configuring NTP/systemd there.

**Comment by jefflill** [comment]
After reading the whole thread for microsoft/WSL#5324, it looks like MSFT has identified this as a Linux kernel bug and has a fix in the works.  Looks like it'll be an updated kernel.

We me need to have neonDESKTOP check the installed WSL kernel and ensure that the correct version is installed although I still hope that this will work seamlessly since we'll be running NTP services.

**Comment by jefflill** [comment]
Microsoft claims to have fixed microsoft/WSL#5324 but people are still having trouble (as are we).  It looks like this is going to be fixed in a new kernel release?

Here are some links to people trying workarounds:

https://github.com/microsoft/WSL/issues/4245#issuecomment-615327939

**Comment by jefflill** [comment]
One quick workaround for unit testing would be for have the Docker related fixtures run this before starting anything:
```
sudo hwclock --hctosys
```

**Comment by jefflill** [comment]
The plit thickens: There may be additional issues for WSL2 running in a VM host, like we're doing for our GitHub runners:

https://tomssl.com/fixing-clock-drift-in-wsl2-using-windows-terminal/

**Comment by jefflill** [comment]
I tried running the new **neon-clocksync** container in `DockerComposeFixture` before starting the application, but that didn't work.

**Comment by jefflill** [comment]
MSFT opened a new issue to track this: https://github.com/microsoft/WSL/issues/10006

**Comment by jefflill** [comment]
**CLOSING:** I'm going to go ahead and close this because it's not impacting us right now; it was impacting Cadence/Temporal client unit testing but we've moved on from being in the Temporal client business.
